arkouda/categorical.py

```diff
@@ -1161,15 +1167,19 @@ def is_registered(self) -> np.bool_:
         Objects registered with the server are immune to deletion until
         they are unregistered.
         """
-        parts_registered: List[np.bool_] = [
-            p.is_registered() for p in Categorical._get_components_dict(self).values()
-        ]
-        if np.any(parts_registered) and not np.all(parts_registered):  # test for error
-            raise RegistrationError(
-                f"Not all registerable components of Categorical {self.name} are registered."
-            )
+        from arkouda.util import is_registered
 
-        return np.bool_(np.any(parts_registered))
+        if self.registered_name is None:
+            result = True
+            result &= is_registered(self.codes.name, as_component=True)
+            result &= is_registered(self.categories.name, as_component=True)
+            result &= is_registered(self._akNAcode.name, as_component=True)
+            if self.permutation is not None and self.segments is not None:
```

Comment:
Yah we no longer give a warning in the case where some but not all components are registered, so it is a change in behavior. But I think just saying no this objects is not registered because some of it's components aren't is a totally reasonable thing to do